### PR TITLE
[6.x] Support for serializers in PhpRedis connection

### PR DESCRIPTION
**Pull request by ReDev1L**
Added support for "none", "php", "igbinary", "msgpack", "json" serializers configuration for PhpRedis connection.

### CODE REVIEW

File: src/Illuminate/Redis/Connectors/PhpRedisConnector.php [hunk 1]
```diff
@@ -91,6 +91,28 @@ protected function createClient(array $config)
             if (! empty($config['read_timeout'])) {
                 $client->setOption(Redis::OPT_READ_TIMEOUT, $config['read_timeout']);
             }
+            if (! empty($config['serializer'])) {
+                $serializers = [
+                    //Redis::SERIALIZER_NONE
+                    'none' => 0,
```
Comment (netpok) [inline src/Illuminate/Redis/Connectors/PhpRedisConnector.php]: Why not use `'none' => Redis::SERIALIZER_NONE`?

For me it looks cleaner.

Comment (GrahamCampbell) [reply src/Illuminate/Redis/Connectors/PhpRedisConnector.php]: I assume due to the extension possibly not being installed?

Comment (ReDev1L) [reply src/Illuminate/Redis/Connectors/PhpRedisConnector.php]: Also it depends on flags, which you used, when you compile ext-redis.
You can compile it w/o igbinary and constant wont be available

Comment (ReDev1L) [reply src/Illuminate/Redis/Connectors/PhpRedisConnector.php]: Refactored to 'defined()' checks, so it will be more straight to understand in code

**Comment by ReDev1L** [comment]
Will need to update documentation, to include 'serializers' section in config database.php:
```
'redis' => [

        'client' => env('REDIS_CLIENT', 'phpredis'),

        'options' => [
            'prefix'  => Str::slug(env('REDIS_KEYS_PREFIX', 'land-objects'), '_'),
            'serializer' => Redis::SERIALIZER_IGBINARY,
            // OR
            'serializer' => 'igbinary', // for easier setting of CI vars
        ],
```
Full available list of values:

- none
- php
- igbinary
- msgpack
- json

- Redis::SERIALIZER_NONE
- Redis::SERIALIZER_PHP
- Redis::SERIALIZER_IGBINARY
- Redis::SERIALIZER_MSGPACK
- Redis::SERIALIZER_JSON

**Comment by GrahamCampbell** [comment]
Maybe we should just have a standard list in Laravel, that converts to the correct stuff at runtime?

**Comment by GrahamCampbell** [comment]
Yeh, I really don't like having all those options that are system dependent. We should just support the textual options, and convert to the constants behind the scenes if we need to.

**Comment by ReDev1L** [comment]
We can skip in_array check for constants. But better to remain defined checks, so user will get error if he doesnt has proper serializer, that he used in config. If we skip them - there is no good informative way to detect error, phpredis extesion will remain silent, if serializer specified is not available

**Comment by GrahamCampbell** [comment]
I'm not suggesting that. I'm suggesting we don't allow the user to write any of these, and we define our own for the purposes of the Laravel config file. It's then up to Laravel to decide how to map them. That way, we can keep the config file independent of the specific PHP setup.

**Comment by netpok** [comment]
I would still leave the option to the user, we could define a default (if it worth over the current implementation), but leave the option. Like we don't try to decide which redis client we use, we provide a sensible default.

Of course if by definition igbinary > msgpack > ... > none, then it can be just used that way and all above is needless.

**Comment by ReDev1L** [comment]
php-redis extension will use "php" as default, if serializer is not set.

> Of course if by definition igbinary > msgpack > ... > none, then it can be just used that way and all above is needless.

That one is dangerous, there can be "microservices" that use one redis instance, but have different php extensions installed. Retarded case, but still possible =).

**Comment by DarkGhostHunter** [comment]
Is this compatible with serializing engine per redis instance?

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**